pkg/webservice/handlers.go

```diff
@@ -614,6 +616,33 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
 		appsDao = append(appsDao, getApplicationDAO(app))
 	}
 
+	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
```

Comment:
Could we have test for gzip header?

Reply:
Can we try to make it modular? We already thinking of doing it in two more places. We should be able to use the method wherever required.

Reply:
1. Test is missing
2. Pls extract all headers to a `const` section, it's better for maintainability. We have string literals all over the place, eg. `writeHeaders()`. I don't think we need to import a separate library for this, but at least have a single place where all of them are defined. Also do the same for MIME types.

Reply:
By modular I mean create two functions. One that tests for the header content see other review point and one that does all the compress work and takes two parameters and returns nothing. We can use that compress function anywhere.
```
func checkHeader(h http.Header, key, value string) bool
func compress(w http.ResponseWriter, v any)
```
Need to file a follow up jira when we add this to add support for all REST endpoints to support compressed data...